maven-core/src/main/java/org/apache/maven/DefaultMaven.java

```diff
@@ -232,6 +232,13 @@ private MavenExecutionResult doExecute( MavenExecutionRequest request, MavenSess
             return addExceptionToResult( result, e );
         }
 
+        for ( WorkspaceReader workspaceReader : getProjectScopedExtensions( session.getProjects(),
+                                                                            WorkspaceReader.class ) )
+        {
+            repoSession.setWorkspaceReader( ChainedWorkspaceReader.newInstance( repoSession.getWorkspaceReader(),
+                                                                                workspaceReader ) );
+        }
+
         WorkspaceReader reactorWorkspace;
```

Comment:
The `ChainedWorkspaceReader` has a constructor taking an array of WorkspaceReader.  I think it would make sense to use it instead of chaining them one by one.
Btw, I think this also applies to the original call a few lines below:
```
        repoSession.setWorkspaceReader( ChainedWorkspaceReader.newInstance( reactorWorkspace,
                                                                            repoSession.getWorkspaceReader() ) );
```

Reply:
done